### DISCUSSION TITLE
Make LocalAI more ethical by adding docs config options

### DISCUSSION THREAD

**Discussion by Darin755** [mudler/LocalAI#1193] [General]
### Background

As you all may know, there are a lot of controversies with AI. This technology has exploded in capability and popularity but we still don't have much in the way of information on its effects. However, that doesn't mean we shouldn't take the time to think about our usage of AI and where its data comes from. There is also some legal gray areas as AI messes with our understanding of ownership.

### A solution (sort of)

Obviously this problem is very broad and has no right answer. That's why I propose a chart that lists all the models localAI supports and there legal, ethical and moral rankings. Such a chart should list legal risks, training data source and a models factual accuracy. I particularly like nextclouds ranking as found here: https://nextcloud.com/blog/ai-in-nextcloud-what-why-and-how/

Anouther issues with LocalAI is its disregard of free/libre software. It pulls in Nvidia proprietary software automatically and does not ask for user consent. I would like to see a completely FOSS version of localAI that doesn't pull any non-free dependencies at all and is completely free of proprietary blobs. Such a image would be limited to CPUs more than likely but in my mind it would be absolutely worth it.

**Comment by mudler** [comment]
> ### Background
> As you all may know, there are a lot of controversies with AI. This technology has exploded in capability and popularity but we still don't have much in the way of information on its effects. However, that doesn't mean we shouldn't take the time to think about our usage of AI and where its data comes from. There is also some legal gray areas as AI messes with our understanding of ownership.
> 
> ### A solution (sort of)
> Obviously this problem is very broad and has no right answer. That's why I propose a chart that lists all the models localAI supports and there legal, ethical and moral rankings. Such a chart should list legal risks, training data source and a models factual accuracy. I particularly like nextclouds ranking as found here: https://nextcloud.com/blog/ai-in-nextcloud-what-why-and-how/

While I tend to second this, I think it is out of scope for the project, however help is appreciated in that direction and contributions are welcome.

> 
> Anouther issues with LocalAI is its disregard of free/libre software. It pulls in Nvidia proprietary software automatically and does not ask for user consent. I would like to see a completely FOSS version of localAI that doesn't pull any non-free dependencies at all and is completely free of proprietary blobs. Such a image would be limited to CPUs more than likely but in my mind it would be absolutely worth it.

That's not accurate and does not picture the current state: there are standard images without Nvidia blobs which are CPU only and Nvidia images. If Nvidia blobs are in standard images I consider that a bug. ffmpeg 
 for similar reasons  is in separated images.

**Reply by Darin755** [reply]
Maybe I used the wrong image. I'll have another look when I get a chance.

**Comment by Darin755** [comment]
Thanks for moving this to a discussion. I honestly forgot discussions existed when I wrote this issue.
